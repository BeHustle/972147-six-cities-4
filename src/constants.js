const CARD_TYPES = [
  `Apartment`,
  `Private room`
];

const Cities = {
  AMSTERDAM: [52.38333, 4.9]
};

const MAP_ZOOM = 12;

const ICON_SIZE = [30, 30];

const ICON_PATH = `img/pin.svg`;

export {CARD_TYPES, Cities, MAP_ZOOM, ICON_SIZE, ICON_PATH};
