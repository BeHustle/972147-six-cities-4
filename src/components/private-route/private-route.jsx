import React from 'react';
import {Route, Redirect} from 'react-router-dom';
import {connect} from "react-redux";
import AppRoute from '../../routes.js';
import {AuthStatus} from '../../constants.js';
import {getAuthStatus} from '../../reducer/user/user.selectors.js';
import PropTypes from 'prop-types';


const PrivateRoute = ({children, path, exact, authStatus}) =>
  <Route path={path} exact={exact}>
    {authStatus === AuthStatus.AUTH
      ? children
      : <Redirect to={AppRoute.LOGIN}/>}
  </Route>;

PrivateRoute.propTypes = {
  path: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  authStatus: PropTypes.oneOf(Object.values(AuthStatus)),
  children: PropTypes.object
};

const mapStateToProps = (state) => ({
  authStatus: getAuthStatus(state),
});


export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
