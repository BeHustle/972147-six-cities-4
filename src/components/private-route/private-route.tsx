import * as React from 'react';
import {Route, Redirect, RouteProps} from 'react-router-dom';
import {connect} from "react-redux";
import AppRoute from '../../routes';
import {AuthStatus} from '../../constants';
import {getAuthStatus} from '../../reducer/user/user.selectors';

type Props = RouteProps & {
  authStatus: string;
  children: () => React.ReactChildren;
}

const PrivateRoute: React.FunctionComponent<Props> = ({children, path, exact, authStatus}: Props) =>
  <Route path={path} exact={exact}>
    {authStatus === AuthStatus.AUTH
      ? children
      : <Redirect to={AppRoute.LOGIN}/>}
  </Route>;

const mapStateToProps = (state) => ({
  authStatus: getAuthStatus(state),
});


export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
