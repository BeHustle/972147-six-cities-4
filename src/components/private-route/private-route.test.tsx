import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {PrivateRoute} from "./private-route";
import {AuthStatus} from '../../constants';
import AppRoute from '../../routes';
import {Router} from 'react-router-dom';
import {history} from '../../history';

it(`Render private route`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <PrivateRoute
            authStatus={AuthStatus.AUTH}
            path={AppRoute.FAVORITES}
            exact={true}>
            <div />
          </PrivateRoute>
        </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
