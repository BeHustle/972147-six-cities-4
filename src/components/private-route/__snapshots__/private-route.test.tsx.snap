// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render private route 1`] = `null`;
